// thirdweb/dist/esm/react/web/ui/ConnectWallet/screens/Buy/fiat/currencies.js
import { jsx as _jsx } from "react/jsx-runtime";
import { RadiobuttonIcon } from "@radix-ui/react-icons";
import { CADIcon } from "../../../icons/currencies/CADIcon.js";
import { EURIcon } from "../../../icons/currencies/EURIcon.js";
import { GBPIcon } from "../../../icons/currencies/GBPIcon.js";
import { JPYIcon } from "../../../icons/currencies/JPYIcon.js";
import { USDIcon } from "../../../icons/currencies/USDIcon.js";
export const usdCurrency = {
    shorthand: "USD",
    name: "US Dollar",
    symbol: "$",
    icon: USDIcon,
};
export const currencies = [
    usdCurrency,
    {
        shorthand: "CAD",
        name: "Canadian Dollar",
        symbol: "$",
        icon: CADIcon,
    },
    {
        shorthand: "GBP",
        name: "British Pound",
        symbol: "£",
        icon: GBPIcon,
    },
    {
        shorthand: "EUR",
        name: "Euro",
        symbol: "€",
        icon: EURIcon,
    },
    {
        shorthand: "JPY",
        name: "Japanese Yen",
        symbol: "¥",
        icon: JPYIcon,
    },
];
export function getCurrencyMeta(shorthand) {
    return (currencies.find((currency) => currency.shorthand.toLowerCase() === shorthand.toLowerCase()) ?? {
        // This should never happen
        icon: UnknownCurrencyIcon,
        name: shorthand,
        symbol: "$",
        shorthand: shorthand,
    });
}
const UnknownCurrencyIcon = (props) => {
    return _jsx(RadiobuttonIcon, { width: props.size, height: props.size });
};
