// thirdweb/dist/cjs/react/web/ui/ConnectWallet/screens/Buy/fiat/currencies.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.currencies = exports.usdCurrency = void 0;
exports.getCurrencyMeta = getCurrencyMeta;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_icons_1 = require("@radix-ui/react-icons");
const CADIcon_js_1 = require("../../../icons/currencies/CADIcon.js");
const EURIcon_js_1 = require("../../../icons/currencies/EURIcon.js");
const GBPIcon_js_1 = require("../../../icons/currencies/GBPIcon.js");
const JPYIcon_js_1 = require("../../../icons/currencies/JPYIcon.js");
const USDIcon_js_1 = require("../../../icons/currencies/USDIcon.js");
exports.usdCurrency = {
    shorthand: "USD",
    name: "US Dollar",
    symbol: "$",
    icon: USDIcon_js_1.USDIcon,
};
exports.currencies = [
    exports.usdCurrency,
    {
        shorthand: "CAD",
        name: "Canadian Dollar",
        symbol: "$",
        icon: CADIcon_js_1.CADIcon,
    },
    {
        shorthand: "GBP",
        name: "British Pound",
        symbol: "£",
        icon: GBPIcon_js_1.GBPIcon,
    },
    {
        shorthand: "EUR",
        name: "Euro",
        symbol: "€",
        icon: EURIcon_js_1.EURIcon,
    },
    {
        shorthand: "JPY",
        name: "Japanese Yen",
        symbol: "¥",
        icon: JPYIcon_js_1.JPYIcon,
    },
];
function getCurrencyMeta(shorthand) {
    return (exports.currencies.find((currency) => currency.shorthand.toLowerCase() === shorthand.toLowerCase()) ?? {
        // This should never happen
        icon: UnknownCurrencyIcon,
        name: shorthand,
        symbol: "$",
        shorthand: shorthand,
    });
}
const UnknownCurrencyIcon = (props) => {
    return (0, jsx_runtime_1.jsx)(react_icons_1.RadiobuttonIcon, { width: props.size, height: props.size });
};
